src/DynamoCore/Graph/Annotations/AnnotationModel.cs

```diff
@@ -358,20 +325,7 @@ internal void UpdateBoundaryFromSelection()
         /// Group the Models based on Height and Width
         /// </summary>
         /// <returns> the width and height of the last model </returns>
-        private Tuple<Double,Double> CalculateWidthAndHeight()
-        {           
-            var xgroup = Nodes.OrderBy(x => x.X).ToList();
-            var ygroup = Nodes.OrderBy(x => x.Y).ToList();
-            double yheight = ygroup.Last().Height;
 
-            //If the last model is Node, then increase the height so that 
-            //node border does not overlap with the group
-            if (ygroup.Last() is NodeModel)
```

Comment:
This part does not seem to be taken into account in the new solution, or maybe I'm missing it. Could `Nodes` contain something that is not a `NodeModel` (not a node)? What would it be?

Still maybe it should be fine to sum `ExtendYHeight` to give some extra space.

Reply:
yeah... so `Nodes` is a terrible name for this member as it is really a list of `ModelBase` - it is the connotation of `Nodes` and `Notes` (little grey boxes containing only text).

I think this code is trying to give differing offsets for various children types and removing this code as @mmisol said is doing away with that check.

Can you please make sure to test these changes with `Notes` included in the group? (annotation)

Reply:
see:
https://github.com/DynamoDS/Dynamo/blob/bca65e58c4be124a3b21f1a648909ca585273425/src/DynamoCore/Graph/Annotations/AnnotationModel.cs#L139

Reply:
![image](https://user-images.githubusercontent.com/66637541/86697213-8f8c3280-bfe4-11ea-8625-bca4a9992621.png)
![note](https://user-images.githubusercontent.com/66637541/86697263-9b77f480-bfe4-11ea-9186-aab79af7a605.PNG)
I just tried it and it works fine with notes too, the only difference being that if the lowest "node" is in fact a note, then the margin will be 5 pixels bigger (thats the hard-coded value of ExtendYHeight). The difference is barely noticeable but it could be fixed by checking if the node is a NodeModel before adding ExtendYHeight to the annotation height.

Reply:
I would be fine with approving this as is, but maybe you can fix this if it is not much work @Astul-Betizagasti . What do you think @mjkkirschner  @QilongTang ?

Reply:
Agree @Astul-Betizagasti Please fix and we will cover the testing part as a team